zebra-chain/src/block/height.rs

```diff
@@ -41,6 +43,54 @@ impl Height {
     pub const MAX_AS_U32: u32 = Self::MAX.0;
 }
 
+// Block heights live in a torsor, they can only be contructed but not computed directly.
+// Addition and substraction is done in the underlying types(u32) and not in the heights themselves.
+// This u32 numbers are a group and they can be added and substracted normally to get a new integer.
+// Having the result of the computation the `Height(X)` constructor can be used to build a new `Height`.
```

Comment:
:+1: This is a good comment to keep, because it explains why the impls are the way they are, using information that isn't already in the source code. However, it might be useful to show to people who are using the code, not just people who are reading it.  So the content could be moved above into the doc comment for the `Height` type above in lines 5-9, maybe as a new paragraph like so:

```
/// The height of a block is the length of the chain back to the genesis block.
///
/// Block heights can't be added, but they can be *subtracted*, to get a difference of block heights, represented as an `i32`, and height differences can be added to block heights to get new heights.
///
/// # Invariants
///
/// Users should not construct block heights greater than `Height::MAX`.

Reply:
Changed to this at https://github.com/ZcashFoundation/zebra/pull/1057/commits/4e3a77479985a2bce82a4cd9fb0d138ecda2fca4